src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -298,19 +343,33 @@ public void RunClassInitialize(TestContext testContext)
         /// <returns>
         /// Any exception that can be thrown as part of a class cleanup as warning messages.
         /// </returns>
-        [SuppressMessageAttribute("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
+        [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Requirement is to handle all kinds of user exceptions and message appropriately.")]
         public string RunClassCleanup()
         {
-            if (this.ClassCleanupMethod == null)
+            if (this.ClassCleanupMethod is null)
             {
-                return null;
+                if (this.BaseClassCleanupMethodsQueue.Count == 0)
```

Comment:
similar to the above. .Any() and can merge with the if on top.

Reply:
Got it, done!